cmake/Modules/G4InstallData.cmake

```diff
@@ -404,17 +404,23 @@ function(geant4_install_dataset _name _destination _timeout)
 
   # - Dispatch to ExternalProject or our own implementation.
   # Use of URL_MD5 *and* TIMEOUT require CMake 2.8.2 or higher.
-  include(ExternalProject)
-  ExternalProject_Add(${_name}
-    PREFIX Externals/${_name}-${_ds_version}
-    SOURCE_DIR ${GEANT4_BUILD_FULL_DATADIR}/${_ds_dir}
-    URL ${_ds_url}
-    URL_MD5 ${_ds_md5sum}
-    TIMEOUT ${_timeout}
-    CONFIGURE_COMMAND ""
-    BUILD_COMMAND ""
-    INSTALL_COMMAND ""
-    )
+  include(FetchContent)
+#  ExternalProject_Add(${_name}
+#    PREFIX Externals/${_name}-${_ds_version}
+#    SOURCE_DIR ${GEANT4_BUILD_FULL_DATADIR}/${_ds_dir}
+#    URL ${_ds_url}
+#    URL_MD5 ${_ds_md5sum}
+#    TIMEOUT ${_timeout}
+#    CONFIGURE_COMMAND ""
+#    BUILD_COMMAND ""
+#    INSTALL_COMMAND ""
+#    )
+  FetchContent_Declare(
+          ${_name}
+          URL          ${_ds_url}
+          URL_HASH     MD5=${_ds_md5sum} # release-1.10.0
+  )
+  FetchContent_MakeAvailable(${_name})
```

Comment:
I'm sorry, but we won't accept this change. Because the data are _not_ required for the build, the download and install as part of it are just there as conveniences. `FetchContent` runs as part of the configure stage, and as it can take time to download the datasets, this will _massively_ increase the configure time which is not acceptable given that the data are not needed (like, say, an external package might be). `ExternalProject` isn't ideal, but it can happily run in parallel in the background of the build whilst the rest is compiling away.

If your use case requires the presence of the data _before_ building, then your project will need to handle this itself.

Reply:
I that with `ExternalProject` there is no target (e.g. `make download`) to download this data and have it available when needed. What about using [ExternalData](https://cmake.org/cmake/help/latest/module/ExternalData.html). Then we can do `ExternalData_Add_Target(data_target)` and `add_dependencies(my_target data_target)`. This way the data is downloaded during build.